### DISCUSSION TITLE
TSDZ2-v20.1C.6  experience

### DISCUSSION THREAD

**Discussion by dzid26** [emmebrusa/TSDZ2-Smart-EBike-1#101] [General]
- So far I think I can tell the motor is a bit quieter at higher frequencies. 
- Smooth start is decent, but of course still a lot of lag. A description of this top of the line ebike that I just watched fits here well https://youtu.be/AnIZ9Dtnb2s?t=283
- ~~Overcurrent~~ feature needs improvement https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/issues/99

Code wise, I appreciate some improvements, but I loathe amount of assembly code added.  By using HAL interrupts, the motor loop execution time should not as critical so I don't really understand the need for the assembly. 🤷 Hopefully it will never need to change, because probably no one will have time to figure it out...

**Comment by KILLERD0G** [comment]
Hi, just tried this firmware on my tsdz2 250w 36v, seems to work well apart from the e07 code. If anyone wants to exchange setting profiles, just let me know. paulrobson@ntlworld.com

**Reply by dzid26** [reply]
Try increasing "battery power" to some high value and instead rely on battery current limit.

**Comment by KILLERD0G** [comment]
Hey buddy, great to hear from you. I really am enjoying the work you 
have done with this kit. Youll have to forgive me but im so new to all 
this and im not sure what you mean. I would not know what to increase or 
to what level. Im really happy to try different profiles out if it helps 
the project. Mine is 250w 36v tsdz2 . Hope your Sunday is going well, 
Paul.


------ Original Message ------
From "dzid26" ***@***.***>
To "emmebrusa/TSDZ2-Smart-EBike-1" 
***@***.***>
Cc "KILLERD0G" ***@***.***>; "Comment" 
***@***.***>
Date 23/06/2024 15:06:16
Subject Re: [emmebrusa/TSDZ2-Smart-EBike-1] TSDZ2-v20.1C.6 experience 
(Discussion #101)

**Reply by KILLERD0G** [reply]
I take it you mean increase these two ? 
![Screenshot 2024-06-23 151215](https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/assets/172324716/f30cbec3-5b1e-478b-b81d-7ffa201caa68)

**Reply by dzid26** [reply]
Leave the current limit as is.  Increase Battery Power Max to 800W.   This is just for the test - it may not help, as I haven't [figure out](https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/issues/99) what exactly is the root cause (for me).

**Reply by KILLERD0G** [reply]
Right, will do. I did use the very latest Java configurator but my motor locked up and eo7 error all the time... Will set this new parameter and get back to you.... Should i use the field weakening  ? Just remember my motor apparently is 36v 250w..

**Reply by dzid26** [reply]
Yes, you can use field weakening.  Even better effect would be if you installed 48V battery with 36V motor.

**Comment by KILLERD0G** [comment]
Hi, would not installing a 48v battery with a 36v motor overheat the 
motor. Shame the batts are so expensive. Would a 52 volts battery work 
just as well ? Doing some testing next week, been a little ill. Please 
keep up the good work. Cheers, Paul.


------ Original Message ------
From "dzid26" ***@***.***>
To "emmebrusa/TSDZ2-Smart-EBike-1" 
***@***.***>
Cc "KILLERD0G" ***@***.***>; "Comment" 
***@***.***>
Date 27/06/2024 19:19:23
Subject Re: [emmebrusa/TSDZ2-Smart-EBike-1] TSDZ2-v20.1C.6 experience 
(Discussion #101)

**Reply by dzid26** [reply]
Voltage doesnt "create" heat.  Current does. But current is controlled so there is no extra heat.  52V would work, but might not needed.

**Comment by KILLERD0G** [comment]
Im hoping support for the tsdz8 comes. My friend just bought one and 
said its so much better than the tsdz2 or 2b...


------ Original Message ------
From "dzid26" ***@***.***>
To "emmebrusa/TSDZ2-Smart-EBike-1" 
***@***.***>
Cc "KILLERD0G" ***@***.***>; "Comment" 
***@***.***>
Date 28/06/2024 09:27:31
Subject Re: [emmebrusa/TSDZ2-Smart-EBike-1] TSDZ2-v20.1C.6 experience 
(Discussion #101)

**Comment by dzid26** [comment]
@emmebrusa  I have an impression that field weakening doesn't work at all since the v20.1C.6 release. Did you get similar reports?  
Above the base speed, the current becomes 1.6A and the torque drops steeply.

**Comment by emmebrusa** [comment]
@dzid26 In the summer I take a break, I don't follow github and forums much.
I would add that in this period I only use the mountain bike with 860C.
In previous versions when the duty cycle reaches 100% and field weakening is activated, the angle increases up to +6 regardless of speed.
A very rapid speed increase is obtained, the torque decrease is equally rapid with a consequent decrease in the duty cycle.
This leads to a oscillation of the motor power.
In the new version for stock displays I made a change, the increase in the angle is proportional to the number of revolutions in a range between 90rpm and 120rpm,
Now with 100% duty cycle and high cadence the assistance is more stable.
I only tried it on the trainer and it seemed to work.
I don't remember making any other changes after that.
I will still have to complete the tests.

**Reply by Bini-99** [reply]
Error TSDZ2-Smart-EBike-1-20.1C.6 with VLCD5
Setting: Hybrid
Odometer:  compensation on 
Setting:  Sport
I am delighted with the OSF software except for these errors:
After 15-30 minutes, the VLCD5 display switches continuously to 85.0 00 and xx
The battery display (max. 6 bars) no longer changes (always 4 bars) 
Press LED on > power is reduced.
A pause is inserted (Speed =00):  The VLCD5 displays correctly again; when driving on, Speed is displayed a few km 00 (compensation).
If Power (off/on) is switched, the display functions normally again. The battery display now shows the correct value (2 bars).
I would like the battery voltage to be displayed when switching on and then only as the original display.
As there are 5 bikes with a VLCD5 display in operation in my area, I do not want to use any other display.

**Comment by wimwes** [comment]
> * So far I think I can tell the motor is a bit quieter at higher frequencies.
> 
>     * Smooth start is decent, but of course still a lot of lag. A description of this top of the line ebike that I just watched fits here well https://youtu.be/AnIZ9Dtnb2s?t=283
> 
>     * ~Overcurrent~ feature needs improvement [E07 overcurrent shows up often #99](https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/issues/99)
> 
> 
> Code wise, I appreciate some improvements, but I loathe amount of assembly code added. By using HAL interrupts, the motor loop execution time should not as critical so I don't really understand the need for the assembly. 🤷 Hopefully it will never need to change, because probably no one will have time to figure it out...

I bought an TSDZ8 it uses J link and no st link. I am not a specialist in these link but I suppose they are not the same and ishould buy an J link. Can de the same software be used? Thnaks for your answer. I am very satisfied with the OSF software on a tsdz2b.

**Reply by dzid26** [reply]
https://github.com/emmebrusa/TSDZ2-Smart-EBike-1/issues?q=is%3Aissue+tsdz8

**Comment by wimwes** [comment]
who can help me a step futher.

**Comment by wimwes** [comment]
Thanks I know what I have to do.


Op do 5 sep. 2024 07:02 schreef Radek ***@***.***>:
